spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/serviceregistry/EurekaServiceRegistryTests.java

```diff
@@ -130,4 +141,54 @@ void eurekaClientGetStatusNoInstance() {
 		assertThat(map).hasSize(1).containsEntry("status", UNKNOWN.toString());
 	}
 
+	@Test
+	void eurekaClientInitializesClientAsynchronously() {
+		when(eurekaInstanceConfigBean.isAsyncClientInitialization()).thenReturn(true);
+		EurekaServiceRegistry registry = new EurekaServiceRegistry(eurekaInstanceConfigBean);
+
+		final AtomicBoolean applicationsFetched = new AtomicBoolean();
+
+		CloudEurekaClient eurekaClient = mock(CloudEurekaClient.class);
+		when(eurekaClient.getApplications()).thenAnswer((answer) -> {
+			applicationsFetched.set(true);
+			return answer;
+		});
+
+		ApplicationInfoManager applicationInfoManager = mock(ApplicationInfoManager.class);
+
+		when(applicationInfoManager.getInfo()).thenReturn(mock(InstanceInfo.class));
+
+		EurekaRegistration registration = EurekaRegistration
+				.builder(new EurekaInstanceConfigBean(new InetUtils(new InetUtilsProperties()))).with(eurekaClient)
+				.with(applicationInfoManager).with(new EurekaClientConfigBean(), mock(ApplicationEventPublisher.class))
+				.with(new SimpleObjectProvider<>(null)).build();
+
+		registry.register(registration);
+
+		await().until(applicationsFetched::get);
```

Comment:
This looks much better, but please also use `Awaitility`'s methods, such as `.atMost(...)` and `pollInterval()` to add additional constraints and ensure frequent polling (so the wait will end if the condition has been met before).

Reply:
Done.